autoload/searchx.vim

```diff
@@ -260,20 +265,28 @@ function! s:on_input(...) abort
     let @/ = l:input
     call searchx#searchundo#searchforward(s:state.direction)
 
+
     " Update view state.
     let s:state.matches = s:find_matches(@/, getcurpos()[1:2])
     call s:refresh({ 'marker': g:searchx.auto_accept })
 
     " Search off-screen match.
     if empty(s:state.matches.matches)
+      echom "no matches"
+      echom s:state.direction
+
       silent noautocmd call winrestview(s:state.firstview)
       let l:next_pos = searchpos(@/, s:state.direction == s:Direction.Next ? 'zn' : 'bn')
       if l:next_pos[0] == 0
-        let s:state.direction = s:state.direction == s:Direction.Next ? s:Direction.Prev : s:Direction.Next
+        " let s:state.direction = s:state.direction == s:Direction.Next ? s:Direction.Prev : s:Direction.Next
```

Comment:
Why was this logic included in the original? This seems to cause a problem where if I'm searching in the forward direction and I type something that finds no results, the direction changes and then I'm sometimes stuck in the reverse direction. If I remove it, it behaves as I would expect. Not sure what the purpose of this is, would love to know!